reference-implementation/lib/writable-stream.js

```diff
@@ -173,11 +176,26 @@ function WritableStreamFinishClose(stream) {
   defaultWriterClosedPromiseResolve(stream._writer);
 }
 
-function WritableStreamFulfillWriteRequest(stream) {
-  assert(stream._writeRequests.length > 0);
+function WritableStreamRejectUnresolvedPromises(stream) {
```

Comment:
The semantics are a bit hard to explain. It's basically "reject everything except .ready and maybe one write()". I couldn't come up with a catchy name for it. But it will be easy to change the name if someone comes up with a better one.

I added the assert.